wandb/sdk/lib/apikey.py

```diff
@@ -200,7 +205,7 @@ def write_netrc(host, entity, key):
 
 def write_key(settings, key, api=None, anonymous=False):
     if not key:
-        return
+        raise ValueError("No API key specified.")
```

Comment:
as long as you have gone through all the paths that could call this and they make sense...  raising a exception seems like a much better thing to do, im just worried there is some path that is calling this with an empty key and expects it to pass

Reply:
all the paths are guarded.  there are 6 in the code:

`configure_api_key()` in `_WandbLogin()` which only is called if `key is not None`, 

and 5 in `prompt_api_key`, which is called only in one place and in that one place has an exception handler (implemented in this PR).